app/scripts/metamask-controller.js

```diff
@@ -2812,6 +2821,20 @@ export default class MetamaskController extends EventEmitter {
    */
   setupControllerEventSubscriptions() {
     let lastSelectedAddress;
+    let lastSelectedSolanaAccountAddress;
```

Comment:
thoughts on using `previousValueComparator` instead of keeping track of the previous value in this var?

Reply:
I'm not sure if its possible given the conditions under which we want to emit the event. If for example the account is changed from solana account 1 to an evm account and then back to solana account 1 we don't want to emit the event... I think we need this piece of state since a single step backwards looking state comparison won't give us the flexibility we need for this logic

Reply:
yup. you right